server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -481,12 +483,20 @@ public boolean containsBrokenAnalysis(String field) {
      */
     public SearchLookup lookup() {
         if (this.lookup == null) {
+            SourceFieldMapper sourceMapper = mappingLookup == null ? null : (SourceFieldMapper) mappingLookup.getMapper("_source");
```

Comment:
Can we make this a new `getSourceProvider()` method on `MappingLookup`?  That way we can easily add in future enhancements like a nested source provider.